litestar/middleware/logging.py

```diff
@@ -191,8 +191,8 @@ def extract_response_data(self, scope: Scope) -> dict[str, Any]:
         connection_state = ScopeState.from_scope(scope)
         extracted_data = self.response_extractor(
             messages=(
-                connection_state.log_context.pop(HTTP_RESPONSE_START),
-                connection_state.log_context.pop(HTTP_RESPONSE_BODY),
+                connection_state.log_context[HTTP_RESPONSE_START],
+                connection_state.log_context[HTTP_RESPONSE_BODY],
```

Comment:
I don't think we have a real need to clear this context out. Firstly, we need the "http.response.start" message to live for as long as we are logging body messages to prevent #3477. And secondly, for every new body message sent, we overwrite the "http.response.body" message in the log context.

The scope will only live as long as the connection, and I'm pretty sure there's no way we could introduce a reference cycle by storing these messages, so I think safe enough to just leave them there.

Reply:
> and I'm pretty sure there's no way we could introduce a reference cycle by storing these messages

Seems reasonable, but do you see an easy way to test this, just to be sure?

Reply:
I'll have a think about it..

The ASGI message would have to have a reference to the scope, or the ScopeState instance, or the log context dict itself to introduce a reference cycle, right?

Reply:
Probably something we should always be wary of with ScopeState. I.e., if we put something in ScopeState that references the scope where the scope state itself is stored, it would be bad for business.

Reply:
> The ASGI message would have to have a reference to the scope, or the ScopeState instance, or the log context dict itself to introduce a reference cycle, right?

Yeah. I'm just not 100% sure this can't happen somewhere. I'll give it a think as well